### PR TITLE
Revert custom sharding WAR

### PR DESCRIPTION
**Pull request by ashors1**

### CODE REVIEW

**Comment by yhtang** [comment]
Is this just a reminder like #477?

**Comment by ashors1** [comment]
> Is this just a reminder like #477?

Yes

**Comment by jaro-sevcik** [comment]
The WAR is not needed anymore since a proper fix landed: https://github.com/openxla/xla/pull/8896

**Comment by olupton** [comment]
Workaround should no longer be needed following https://github.com/openxla/xla/pull/8896. @jaro-sevcik checked this morning.

**Review by olupton — APPROVED** [review]
LGTM, merging before nightlies

File: .github/container/build-jax.sh [hunk 1]
```diff
@@ -133,10 +133,6 @@ while [ : ]; do
             XLA_ARM64_PATCH_LIST=$2
             shift 2
             ;;
-        --other-xla-patches)
```
Comment (nouiz) [inline .github/container/build-jax.sh:136]: Keeping that attribute could be useful in the future.
We can always add it back later, but having it for dev could be useful.

**Pull request merged**